### ISSUE TITLE
Strange behavior with random timeouts, refreshing loads page immediately 

### ISSUE THREAD

**Issue by christarazi** [pi-hole/pi-hole#935]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [] The issue I am reporting can be *replicated*
- [x] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

- [] 1 (very unfamiliar)
- [x] 2
- [] 3
- [] 4
- [] 5
- [] 6
- [] 7
- [] 8
- [] 9
- [] 10 (very familiar)

---

**[ISSUE] Expected Behaviour:**
No timeout when browsing the web.

**[ISSUE] Actual Behaviour:**

I've noticed when pihole receives a DNS request that has no IPv6 resolution, it returns `NODATA-IPv6`. The website which makes this request will hang and in the browser it just looks like it's loading something. If I refresh the page, the web page loads instantly. 

**[ISSUE] Steps to reproduce:**

Kind of hard to reproduce, essentially just find a site that doesn't use IPv6 so that `NODATA-IPv6` is returned. Most likely, it will hang and refreshing makes the page load again.
-
-
-
-

**(Optional) Debug Log generated by `pihole -d`:**

http://termbin.com/c3uo

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

I'm not sure what could be causing this. If this is an issue with `dnsmasq` or `pihole`. I have attached a termbin log, the issue starts with making a request to stackoverflow. You'll see the `NODATA-IPv6` and then several seconds later, I force refresh and it loads fine. This is not unique to stackoverflow. It happens on any site. One specifically I remember is reddit and sometimes it will hang on `*.thumbs.redditmedia.com`.

Please let me know if you need me to clarify the issue or anything else I can do.

**Comment by Mcat12** [comment]
It appears that [Reddit does not support IPv6](https://www.reddit.com/r/ipv6/comments/41jq5h/so_will_reddit_ever_support_ipv6/), though you'd think the browser would handle this without manual intervention.
![Image](http://image.prntscr.com/image/eba86713522e4c11a15c236104b83535.png)
Stackoverflow also appears to [not](http://meta.stackexchange.com/questions/21592/would-it-be-possible-for-stack-overflow-to-be-accessible-over-ipv6) support [IPv6](http://meta.stackexchange.com/questions/134740/stack-overflow-over-ipv6-on-world-ipv6-launch-day). There are many more sites that are only on IPv4, so this may be a browser issue. Dnsmasq does return the `NODATA-IPv6` flag within the same second it got the query (and responds to the IPv4 query as quickly), so that's not the cause of the slowdown.

**Comment by christarazi** [comment]
I don't think it's a browser issue because all the devices in my house exhibited this behaviour. I reverted the DNS responsibilities back to my router. I plan to see if that fixes it, and then only set up my Pi for only my PC.

**Comment by dschaper** [comment]
Are your clients IPv6 enabled and have you told them about the Pi-hole via IPv6 DNS servers? Just checking a few base issues to make sure everything is okay up to the Pi-hole.

**Comment by christarazi** [comment]
Yes all the clients have IPv6 enabled. I configured my router (still doing DHCP duties) to pass along the Pi-hole's IP address, so yes my devices do know about Pi-hole. 

    On Saturday, November 19, 2016 10:37 AM, Dan Schaper <notifications@github.com> wrote:
 

 Are your clients IPv6 enabled and have you told them about the Pi-hole via IPv6 DNS servers? Just checking a few base issues to make sure everything is okay up to the Pi-hole. —

**Comment by dschaper** [comment]
I think the IPv6 may be a red herring, and instead this is an asset issue. It may be the JS that those sites are using is being blocked and the browsers are timing out. We'd need to check on a client machine a site like ipv6-test.com that pulls a full suite of DNS resolver tests to see if there are failures in the 6to4. Do you clients have dual-stack configurations? They should auto-fail over to IPv4 if there are any issues with IPv6.

**Comment by dschaper** [comment]
Below is what I see on a stock install with the router announcing both IPv4 DNS and IPv6 DNS on a dual-stack client with IPv6 enabled.

![untitled](https://cloud.githubusercontent.com/assets/2782850/20458056/f06fdc24-ae4e-11e6-8efb-ccf84eb940e5.png)

**Comment by christarazi** [comment]
So I just reset my router to default settings. My router's defaults are IPv4 everything. I just configured DHCPv6 for IPv6, and I was able to get my PC to dual-stack. I will test out this on my PC **without** Pi-hole to see if my router is the one causing issues.

![image](https://cloud.githubusercontent.com/assets/3885308/20458192/bd18689c-ae52-11e6-9ce0-2f5745c9962e.png)

**Comment by dschaper** [comment]
Great, some routers do require a link-local address for internal DNS, not sure why.

**Comment by christarazi** [comment]
No problems so far. I ran without Pi-hole for a day just to verify the source of the issues. Then the next day, I made sure my router was configured to dual-stack and reinstalled Pi-hole. Been running fantastic for almost a week now. So I guess the conclusion is that I may have mistakenly assumed that I was running dual-stack, but in reality only running IPv4 and that was causing the problems. Thank you very much @dschaper.
